source/extensions/filters/listener/proxy_protocol/proxy_protocol.cc

```diff
@@ -222,6 +222,18 @@ ReadOrParseState Filter::parseBuffer(Network::ListenerFilterBuffer& buffer) {
       return read_header_state;
     }
     if (header_version_ == ProxyProtocolVersion::NotFound) {
+      if (!cb_->filterState().hasData<Network::ProxyProtocolFilterState>(
```

Comment:
This block seems incorrect. You're setting the proxy protocol filter state when there isn't any proxy protocol. `NotFound` seems implied by the absence of this filter state.

Reply:
I would have like t have a ways to differentiate between proxy-protocol filter was present but did not found any proxy-protocol in the connection and they are no proxy-protocol filter. But I am fine with removing it.